mesonbuild/modules/gnome.py

```diff
@@ -1016,55 +1017,73 @@ def yelp(self, state: 'ModuleState', args: T.Tuple[str, T.List[str]], kwargs) ->
         else:
             if len(args) > 1:
                 mlog.warning('"gnome.yelp" ignores positional sources arguments when the "sources" keyword argument is set')
-        source_str = '@@'.join(sources)
+        sources_files = [mesonlib.File.from_source_file(state.environment.source_dir,
+                                                        os.path.join(state.subdir, 'C'),
+                                                        s) for s in sources]
 
         langs = mesonlib.stringlistify(kwargs.pop('languages', []))
+        if not langs:
+            langs = read_linguas(state.subdir)
+        # I am currently ignoring media and symlinks
+        # How could they be added?
```

Comment:
Ok, I'll look into this and see if I have the skills to tackle it. If I feel I could pull it through, then I would rather take care of it now. That it will definitely delay the landing of this PR, but also save work on the long-term. Given how long this helper has been around, I guess there's no real rush to pull this is only half-way cleaned?

Reply:
A more important consideration is that this should only land for meson 0.61, which isn't imminent, so we have time to polish it before release. :)

Reply:
There is a temptative implementation in #9608. I have rebased this PR on top of it just for testing purposes. But seems to work alright on this scenario :)